extension/test/unit/cart/fetchShopifyCheckout.test.js

```diff
@@ -0,0 +1,58 @@
+const sinon = require('sinon')
+const assert = require('assert')
+const rewire = require('rewire')
+const ShopifyApiRequest = require('../../../lib/shopify.api.js')
+
+const fetchShopifyCheckout = rewire('../../../cart/fetchShopifyCheckout.js')
+
+describe('fetchShopifyCheckout', () => {
+  /** @var {StepContext} */
+  const context = {
+    storage: {
+      device: {},
+      user: {
+        get: () => { return 'token' },
+        set: () => {}
+      }
+    },
+    config: {
+      shopifyShopAlias: 'shopgate'
+    },
+    log: {
+      debug: () => {
+      },
+      error: () => {
+      }
+    },
+    meta: {
+      userId: 1
+    }
+  }
+
+  before(() => {
+
+  })
+
+  it('should load the checkout by token', async () => {
+    sinon.stub(ShopifyApiRequest.prototype, 'getCheckout').returns({ checkout: { } })
+
+    const loadCheckoutTokenSpy = sinon.spy(fetchShopifyCheckout.__get__('loadCheckoutToken'))
+    fetchShopifyCheckout.__set__('loadCheckoutToken', loadCheckoutTokenSpy)
+
+    await fetchShopifyCheckout(context, { createNew: false })
+    assert(loadCheckoutTokenSpy.calledOnce, 'loadCheckoutToken should be called once')
+  })
+
+  it('should create a new checkout', async () => {
```

Comment:
you could create some tests that just check what `isNew` flag is returned, at the moment you are not checking for that, just what function is called. Technically, doesn't that mean these tests should pass with the bug as well?

Reply:
I am testing it implicitly with the saveCheckoutToken spy. Also added it to not be called in the first test. saveCheckoutToken should only be called if isNew = true. I guess there is no other way to test to local variable isNew.

Reply:
it's kinda of a wonky solution. That is, you are not testing a unit. A function is a unit, but instead of testing the `fetchCheckout` function to see what it returns, you are testing the whole step and spy on functions inside. This is prone to some problems as someone can modify your insides and the tests can still be green. 

You don't technically need any of this spying stuff to test this function, you just need to sinon stub the return of `loadCheckoutToken` function and test the return.

Reply:
Unit in unit test is not limited to a function per definition, as there is no general definition that defines a unit as a method. It only refers to "part of code", "component" e.g. which could be a method, a class, a step (in our case) that is tested on it's own without 'external' code (=code outside of the set unit) being taken into account. It's just important to mock out external stuff like the shopify request class. 
Just stubbing the loadCheckoutToken is not sufficient if we want to test if a checkout will be created new or only one is loaded. So that test would not detect the issue that caused this bug. But I totally second that even having spies in the unit test implies that the code is not really testable as it is not as robust as a unit test should be.